### PR TITLE
dm-worker: support election with same source-id

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 

support election with the same source-id for dm-worker

### What is changed and how it works?

1. add dm-masters' address config for dm-worker
2. use pkg/election to election leader on `source-id`

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: dm/master/config.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func (c *Config) adjust() error {
 		if err != nil {
 			return terror.ErrMasterGetHostnameFail.Delegate(err)
 		}
-		c.Name = fmt.Sprintf("%s-%s", defaultNamePrefix, hostname)
+		c.Name = fmt.Sprintf("%s-%s-%s", defaultNamePrefix, hostname, port)
```
Comment (csuzhangxc) [inline dm/master/config.go]: 👍

File: dm/worker/worker.go [hunk 2]
```diff
@@ -68,6 +79,11 @@ type Worker struct {
 	tracer *tracing.Tracer
 
 	taskStatusChecker TaskStatusChecker
+
+	etcdClient *etcd.Client
+	election   *election.Election
+
+	name string
```
Comment (WangXiangUSTC) [inline dm/worker/worker.go]: useless, I will remove this line next commit

**Pull request closed without merging**